src/flb_pack.c

```diff
@@ -329,90 +329,90 @@ static int msgpack2json(char* buf, int *off, size_t left, msgpack_object *o)
     case MSGPACK_OBJECT_POSITIVE_INTEGER:
         {
             char temp[32] = {0};
-            snprintf(temp, 31, "%lu", (unsigned long)o->via.u64);
-            ret = try_to_write(buf, off, left,temp,0);
+            ret = try_to_write(buf, off, left, temp,
+                snprintf(temp, 31, "%lu", (unsigned long)o->via.u64));
         }
         break;
 
     case MSGPACK_OBJECT_NEGATIVE_INTEGER:
         {
             char temp[32] = {0};
-            snprintf(temp, 31, "%ld", (signed long)o->via.i64);
-            ret = try_to_write(buf,off,left,temp,0);
+            ret = try_to_write(buf, off, left, temp,
+                 snprintf(temp, 31, "%ld", (signed long)o->via.i64));
```

Comment:
- since you are using temp[32] in different places, declare it for the whole function.
- please do not add snprintf() inside try_to_write() as parameter. do it in a separate line, that's fine.
- for snprintf buf length parameter, use sizeof(temp) - 1.